### PR TITLE
Add test for maintenance_period. closes #808

### LINKED ISSUE #808: Add some tests for maintenance period
Add some tests for the maintenance period for hosts and services

### PR DESCRIPTION
**Pull request by ddurieux**

### CODE REVIEW

File: test/test_maintenance_period.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        timeperiod = Timeperiod(data)
+        timeperiod.explode()
+        self.schedulers['scheduler-master'].sched.timeperiods[timeperiod.uuid] = timeperiod
+        host.maintenance_period = timeperiod.uuid
+
+        # Make the host be UP again
+        self.scheduler_loop(1, [[host, 0, 'UP']])
+
+        assert 1 == len(host.downtimes)
+        # The service is still in a downtime period
```
Comment (mohierf) [inline test/test_maintenance_period.py]: "The host is in a downtime period" looks better

File: test/test_maintenance_period.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        assert host.in_scheduled_downtime
+        downtime = host.downtimes.values()[0]
+        assert downtime.fixed
+        assert downtime.is_in_effect
+        assert not downtime.can_be_deleted
+        assert downtime.end_time == timestamp_end
+
```
Comment (mohierf) [inline test/test_maintenance_period.py]: Is there any specific downtime comment for the maintenance period? If so, we should test this comment value, no?

Comment (ddurieux) [inline test/test_maintenance_period.py]: you're right, I miss it

File: test/test_maintenance_period.py (hunk 1 above)
Comment (ddurieux) [inline test/test_maintenance_period.py]: :D

**Comment by mohierf** [comment]
Please squash the commits and I will merge this.

**Pull request merged**